### PR TITLE
gate fetchDepositAddress network support

### PR DESCRIPTION
**Pull request by ndubel**
`networksById` in `options` looks unused as for now so I move it to `safeNetwork` with actual changes

### CODE REVIEW

File: ts/src/gate.ts [hunk 1]
```diff
@@ -606,26 +606,6 @@ export default class gate extends Exchange {
                     'SOLANA': 'SOL',
                     'TRC20': 'TRX',
                 },
-                'networksById': {
```
Comment (carlosmiei) [inline ts/src/gate.ts:609]: Actually this is used by `networkIdToCode` and `networkCodeToId` but yeah this object is filled automatically from the `networks`

File: ts/src/gate.ts [hunk 2]
```diff
@@ -973,6 +953,18 @@ export default class gate extends Exchange {
         return super.safeMarket (marketId, market, delimiter, marketType);
     }
 
+    safeNetwork (networkId) {
```
Comment (carlosmiei) [inline ts/src/gate.ts:956]: instead of safeNetwork, we should use `networkIdToCode` and `networkCodeToId`

Comment (ndubel) [reply ts/src/gate.ts]: done, I've also remove some strange mappings

Comment (carlosmiei) [reply ts/src/gate.ts]: @ndubel thanks will check it

**Comment by carlosmiei** [comment]
DEMO
```
p gate fetchDepositAddress USDT
Python v3.11.5
CCXT v4.1.85
gate.fetchDepositAddress(USDT)
{'address': 'XXX',
 'code': 'USDT',
 'currency': 'USDT',
```
```
p gate fetchDepositAddress USDT '{"network": "TRC20"}'
Python v3.11.5
CCXT v4.1.85
gate.fetchDepositAddress(USDT,{'network': 'TRC20'})
{'address': 'XX',
 'code': 'USDT',
 'currency': 'USDT',
'network': 'TRC20',
 'tag': None}
```

**Pull request merged**